plugins/thumbnail/lib/model/thumbnailFactory/kImageTransformationAdapter.php

```diff
@@ -345,16 +333,22 @@ protected function handleStripProfile($step)
 	}
 
 	/**
-	 * @param bool $bestFit
 	 * @param kImageTransformationStep $step
+	 * @param $width
+	 * @param $height
+	 * @param bool $bestFit
 	 */
-	protected function handleResize($bestFit, $step)
+	protected function addResizeAction($step, $width, $height, $bestFit = false)
 	{
 		$action = new kResizeAction();
-		$action->setActionParameter(kThumbnailParameterName::WIDTH, $this->parameters->get(kThumbFactoryFieldName::WIDTH));
-		$action->setActionParameter(kThumbnailParameterName::HEIGHT, $this->parameters->get(kThumbFactoryFieldName::HEIGHT));
-		$action->setActionParameter(kThumbnailParameterName::BEST_FIT, $bestFit);
-		$step->addAction($action);
+		$bestFit = $bestFit && ($this->parameters->get(kThumbFactoryFieldName::WIDTH) > kResizeAction::BEST_FIT_MIN && $this->parameters->get(kThumbFactoryFieldName::HEIGHT) > kResizeAction::BEST_FIT_MIN);
```

Comment:
why you use for best fit $this->parameters->get(kThumbFactoryFieldName::WIDTH) and $this->parameters->get(kThumbFactoryFieldName::HEIGHT) and not using the height and width params of the functions?

Reply:
Good catch